docs/source/specification.rst

```diff
@@ -0,0 +1,30 @@
+.. _specification:
+
+====================================================
+The Reproducible Execution Environment Specification
+====================================================
+
+repo2docker scans a repository for particular :ref:`config-files`, such
+as ``requirements.txt`` or ``REQUIRE``. The collection of files, their contents,
+and the resulting actions that repo2docker takes is known
+as the **Reproducible Execution Environment Specification** (or REES).
+
+The goal of the REES is to automate and encourage existing community best practices
+for reproducible computational environments. This includes installing
+community-standard specification files such as ``requirements.txt`` or ``REQUIRE`` using
+standard tools such as ``pip`` or ``conda`` or ``apt``. While repo2docker automates the
+creation of the environment, a human should be able to look at a REES-compliant
+repository and reproduce the environment using common, clear steps without
+repo2docker software.
+
+Currently, the definition of the REE Specification is the following:
+
+    Any collection of files taken from the :ref:`config-files`
+    list, placed either in the root of a folder or in a sub-folder called
+    either ``binder/`` or ``.binder/``.
```

Comment:
Could we turn L20-24 into something like "Any directory containing zero or more files from the config-files list is a valid *reproducible execution environment* as defined by the REES. The configuration files have to all be placed either in the root of the directory, in a `binder/` sub-directory or a `.binder/` sub-directory."?

In my mind a specification is a document that describes a thing which it defines. What I was expecting is something like "The HTTP spec defines that a HTTP request consists of a header and possibly a body. The header is separated from the body by two `\n`. A valid header consists of zero or more lines ... etc" https://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html Something like this is what I think of when I think of REES.

Reply:
I think eventually we should have that kind of more specific language in there, I agree